onnxruntime/core/providers/cpu/math/einsum_utils.cc

```diff
@@ -0,0 +1,724 @@
+    // as most likely bradcasted dims will be batch dimensions (i.e.) outermost dimensions and hence we don't have to pay
+    // transposing while "homogenizing" the input
+
+    // Hence offset all subscript indices by num_of_ellipsis_dims_
+    for (size_t i = 0; i < EinsumOp::num_of_letters; ++i) {
+      if (letter_to_index_[i] != -1) {
+        letter_to_index_[i] += num_of_ellipsis_dims_;
+      }
+    }
+
+    std::vector<int64_t> temp_index_to_last_input(num_subscript_indices_, -1);
+    for (size_t i = 0; i < subscript_indices_to_last_input_.size(); ++i) {
+      temp_index_to_last_input[i + num_of_ellipsis_dims_] = subscript_indices_to_last_input_[i];
+    }
+    subscript_indices_to_last_input_ = std::move(temp_index_to_last_input);
+
+    std::vector<int64_t> temp_index_to_dim_value(num_subscript_indices_, -1);
+    for (size_t i = 0; i < subscript_indices_to_dim_value_.size(); ++i) {
+      temp_index_to_dim_value[i + num_of_ellipsis_dims_] = subscript_indices_to_dim_value_[i];
+    }
+    subscript_indices_to_dim_value_ = std::move(temp_index_to_dim_value);
+
+    for (size_t i = 0; i < input_subscript_indices_.size(); ++i) {
+      auto& current_input_dim_indices_to_subscript_indices = input_subscript_indices_[i];
+      std::vector<int64_t> temp_current_input_dim_indices_to_subscript_indices;
+      temp_current_input_dim_indices_to_subscript_indices.reserve(current_input_dim_indices_to_subscript_indices.size());
+
+      const auto& dims = inputs_[i]->Shape().GetDims();
+      auto rank = dims.size();
+
+      size_t dim_iter = 0;
+      size_t num_broadcasted_indices = 0;
+      while (dim_iter < current_input_dim_indices_to_subscript_indices.size()) {
+        auto value = current_input_dim_indices_to_subscript_indices[dim_iter];
+        if (value == 26) {  //This is a broadcasted dim
+          ORT_ENFORCE(num_broadcasted_indices < num_of_ellipsis_dims_);
+          temp_current_input_dim_indices_to_subscript_indices.push_back(static_cast<int64_t>(num_broadcasted_indices));
+          subscript_indices_to_last_input_[num_broadcasted_indices] = i;
+
+          // This is the first time we are seeing this broadcasted dim
+          if (subscript_indices_to_dim_value_[num_broadcasted_indices] == -1) {
+            subscript_indices_to_dim_value_[num_broadcasted_indices] = dims[dim_iter];
+          } else {  // We have seen this broadcasted dim before
+            // Check if the previous value is equal to the current value
+            if (subscript_indices_to_dim_value_[num_broadcasted_indices] != dims[dim_iter]) {
+              // If they are not equal, one of them needs to be 1
+              if (subscript_indices_to_dim_value_[num_broadcasted_indices] == 1) {
+                subscript_indices_to_dim_value_[num_broadcasted_indices] = dims[dim_iter];
+              } else {
+                ORT_ENFORCE(dims[dim_iter] == 1, "Given inputs are not broadcastable");
+              }
```

Comment:
Need some more info in the error message about which inputs are not broadcastable and why.

Reply:
Done.